File: functions/alsa-lib/variants/go/2_muldiv_down_1.go
package main

func muldiv_down(a int32, b int32, c int32) int32 {
	v := int64(a) * int64(b) / int64(c)
	switch {
	case v > int64(2147483647):
		return 2147483647
	case v < int64(-2147483648):
		return -2147483648
	default:
		return int32(v)
	}
}


func main(){
    muldiv_down(0,0,0)
}

File: functions/alsa-lib/variants/c/2_muldiv_down_1.c
#include <stdint.h>

static inline int muldiv_down(int a, int b, int c)
{
	int64_t v = (int64_t)a * b;
	v = v / c;
	if (v > 0x7FFFFFFF) {
		return 0x7FFFFFFF;
	}
	if (v < -0x80000000) {
		return -0x80000000;
	}
	return (int)v;
}


int main(){
    muldiv_down(0,0,0);
}
